ext/openssl/ossl_pkey.c

```diff
@@ -83,23 +83,69 @@ ossl_pkey_new(EVP_PKEY *pkey)
 # include <openssl/decoder.h>
 
 EVP_PKEY *
-ossl_pkey_read_generic(BIO *bio, VALUE pass)
+ossl_pkey_read(BIO *bio, const char *input_type, int selection, VALUE pass)
 {
     void *ppass = (void *)pass;
-    OSSL_DECODER_CTX *dctx;
+    OSSL_DECODER_CTX *dctx = NULL;
     EVP_PKEY *pkey = NULL;
-    int pos = 0, pos2;
+    int pos = 0, pos2 = 0;
```

Comment:
dctx and pos2 don't need to be initialized here.

Reply:
Thanks for pointing out this> I will fix it. Sorry. I noticed that the local variables are initialized only if these are necessary. However, I tended to initialize every local variables as much as possible as I didn't care about how the variables are used. But I will respect this repository's coding style.

Reply:
I fixed this one.

Reply:
I prefer not to initialize unless necessary because compiler warnings on usage of uninitialized variables can be useful to find bugs.

This is more of a style preference, though, I just didn't see a reason to make diff larger.

Reply:
All right. I can understand it.